### PR TITLE
Fix default values for invalid config.json files

### LINKED ISSUE #2493: Upgrade to 10.7.0 completely broke web interface.
**Describe The Bug**
After upgrading from 10.6.4 the web ui stopped functioning.

**Steps To Reproduce**
Confirmed cases of this happening have been on Debian 9 and Ubuntu 20.04
1. Have an existing installation of jellyfin 10.6.4
2. Upgrade to 10.7.0 (Using apt-get in my case)

**Expected Behavior**
Upgrade succeeds and able to use the web interface like normal

**Logs**
```
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto3c6ca3cf906a3cadc4ef3e8b8647fc80.woff2'. URL: http://media.iqfx.nl/jellyfin/web/auto3c6ca3cf906a3cadc4ef3e8b8647fc80.woff2
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto4111.216eaafc1998676a57da.bundle.js'. URL: http://media.iqfx.nl/jellyfin/web/auto4111.216eaafc1998676a57da.bundle.js
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto4201.842f5b8e2be0c61096be.bundle.js'. URL: http://media.iqfx.nl/jellyfin/web/auto4201.842f5b8e2be0c61096be.bundle.js
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto3d1ccc6a42fa9c614ae5c1a2cd99c728.woff2'. URL: http://media.iqfx.nl/jellyfin/web/auto3d1ccc6a42fa9c614ae5c1a2cd99c728.woff2
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto4258.9967bd8cdf274b59c1e8.bundle.js'. URL: http://media.iqfx.nl/jellyfin/web/auto4258.9967bd8cdf274b59c1e8.bundle.js
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto42a889d51221715273a8d3b9176addcf.woff2'. URL: http://media.iqfx.nl/jellyfin/web/auto42a889d51221715273a8d3b9176addcf.woff2
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto4481.f752f84b494404205367.bundle.js'. URL: http://media.iqfx.nl/jellyfin/web/auto4481.f752f84b494404205367.bundle.js
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto455fafb7b7d43f13c8d3013584231cf4.woff2'. URL: http://media.iqfx.nl/jellyfin/web/auto455fafb7b7d43f13c8d3013584231cf4.woff2
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto46f8949c5873634fd15b162cab886980.woff2'. URL: http://media.iqfx.nl/jellyfin/web/auto46f8949c5873634fd15b162cab886980.woff2
Mar 09 00:59:26 jellyfin jellyfin[588]: [00:59:26] [ERR] Error processing request: Could not find file '/usr/share/jellyfin/web/auto4a10f1826c3782dd9868545050d23101.woff2'. URL: http://media.iqfx.nl/jellyfin/web/auto4a10f1826c3782dd9868545050d23101.woff2
```

**Screenshots**
jellyfinwat

**System (please complete the following information):**
 - Platform: Confirmed cases of this happening have been on Debian 9 and Ubuntu 20.04
 - Browser: Any
 - Jellyfin Version: 10.7.0

**Additional Context**
I'm using the /jellyfin sub url if that makes any difference.

### LINKED ISSUE #2494: Black screen 10.7.0
**Describe The Bug**

Updated to version 10.7.0 and both the black screen appears in iOS and Web.

**Steps To Reproduce**

Black screen
**Expected Behavior**

**Logs**

**Screenshots**

**System (please complete the following information):**
 - Platform: [Ubuntu 18.04.5 LTS server ]
 - Browser: [Firefox, iOS app]
 - Jellyfin Version: [10.7.0]

**Additional Context**

Updated from terminal with sudo apt update and apt upgrade and default option retain “config.json”

### LINKED ISSUE #2500: Completely empty interdace...
**Describe The Bug**
My Jellifyn...
image

**Steps To Reproduce**

1. Go to any side in jellyfin

**Expected Behavior**
Really!!

**Logs**
image
[2021-03-10 01:49:06.365 +01:00] [INF] User Changes Sync Finished Taking "00:00:00.0037677"
[2021-03-10 01:50:58.402 +01:00] [WRN] WS "192.0.2.128" error receiving data: "The remote party closed the WebSocket connection without completing the close handshake."
[2021-03-10 01:50:58.410 +01:00] [INF] WS "192.0.2.128" closed
[2021-03-10 01:51:25.751 +01:00] [INF] WS "192.0.2.128" request
[2021-03-10 01:52:13.781 +01:00] [INF] Sending ForceKeepAlive message to 1 inactive WebSockets.
[2021-03-10 02:28:21.475 +01:00] [INF] Jellyfin version: "10.7.0"
[2021-03-10 02:28:21.544 +01:00] [INF] Environment Variables: ["[JELLYFIN_DATA_DIR, /var/lib/jellyfin]", "[JELLYFIN_LOG_DIR, /var/log/jellyfin]", "[JELLYFIN_RESTART_OPT, --restartpath=/usr/lib/jellyfin/restart.sh]", "[JELLYFIN_CONFIG_DIR, /etc/jellyfin]", "[JELLYFIN_ARGS, $JELLYFIN_WEB_OPT $JELLYFIN_RESTART_OPT $JELLYFIN_FFMPEG_OPT $JELLYFIN_SERVICE_OPT $JELLYFIN_NOWEBAPP_OPT]", "[JELLYFIN_USER, jellyfin]", "[JELLYFIN_FFMPEG_OPT, --ffmpeg=/usr/lib/jellyfin-ffmpeg/ffmpeg]", "[JELLYFIN_WEB_OPT, --webdir=/usr/share/jellyfin/web]", "[JELLYFIN_CACHE_DIR, /var/cache/jellyfin]"]
[2021-03-10 02:28:21.549 +01:00] [INF] Arguments: ["/usr/lib/jellyfin/bin/jellyfin.dll", "--webdir=/usr/share/jellyfin/web", "--restartpath=/usr/lib/jellyfin/restart.sh", "--ffmpeg=/usr/lib/jellyfin-ffmpeg/ffmpeg", ""]
[2021-03-10 02:28:21.551 +01:00] [INF] Operating system: "Linux"
[2021-03-10 02:28:21.551 +01:00] [INF] Architecture: X64
[2021-03-10 02:28:21.552 +01:00] [INF] 64-Bit Process: True
[2021-03-10 02:28:21.552 +01:00] [INF] User Interactive: True
[2021-03-10 02:28:21.552 +01:00] [INF] Processor count: 8
[2021-03-10 02:28:21.552 +01:00] [INF] Program data path: "/var/lib/jellyfin"
[2021-03-10 02:28:21.553 +01:00] [INF] Web resources path: "/usr/share/jellyfin/web"
[2021-03-10 02:28:21.553 +01:00] [INF] Application directory: "/usr/lib/jellyfin/bin/"
[2021-03-10 02:28:21.978 +01:00] [INF] Setting cache path: "/var/cache/jellyfin"
[2021-03-10 02:28:22.242 +01:00] [INF] Loading assemblies
[2021-03-10 02:28:22.264 +01:00] [INF] Loaded assembly "Jellyfin.Plugin.CoverArtArchive, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Cover Art Archive_2.0.0.0/Jellyfin.Plugin.CoverArtArchive.dll"
[2021-03-10 02:28:22.265 +01:00] [INF] Loaded assembly "Jellyfin.Plugin.Fanart, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Fanart_6.0.0.0/Jellyfin.Plugin.Fanart.dll"
[2021-03-10 02:28:22.269 +01:00] [INF] Loaded assembly "LiteDB, Version=4.1.4.0, Culture=neutral, PublicKeyToken=4ee40123013c9f27" from "/var/lib/jellyfin/plugins/Kodi Sync Queue_6.0.0.0/LiteDB.dll"
[2021-03-10 02:28:22.274 +01:00] [INF] Loaded assembly "Jellyfin.Plugin.KodiSyncQueue, Version=6.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Kodi Sync Queue_6.0.0.0/Jellyfin.Plugin.KodiSyncQueue.dll"
[2021-03-10 02:28:22.276 +01:00] [INF] Loaded assembly "Jellyfin.Plugin.PlaybackReporting, Version=9.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Playback Reporting_9.0.0.0/Jellyfin.Plugin.PlaybackReporting.dll"
[2021-03-10 02:28:22.280 +01:00] [INF] Loaded assembly "Jellyfin.Plugin.Reports, Version=11.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/Reports_11.0.0.0/Jellyfin.Plugin.Reports.dll"
[2021-03-10 02:28:22.281 +01:00] [INF] Loaded assembly "Jellyfin.Plugin.TMDbBoxSets, Version=7.0.0.0, Culture=neutral, PublicKeyToken=null" from "/var/lib/jellyfin/plugins/TMDb Box Sets_7.0.0.0/Jellyfin.Plugin.TMDbBoxSets.dll"
[2021-03-10 02:28:28.261 +01:00] [INF] Loaded plugin: "Cover Art Archive" "2.0.0.0"
[2021-03-10 02:28:28.263 +01:00] [INF] Loaded plugin: "Fanart" "6.0.0.0"
[2021-03-10 02:28:28.265 +01:00] [INF] Jellyfin.Plugin.KodiSyncQueue is now starting
[2021-03-10 02:28:28.265 +01:00] [INF] Creating DB Repository...
[2021-03-10 02:28:28.284 +01:00] [INF] Loaded plugin: "Kodi Sync Queue" "6.0.0.0"
[2021-03-10 02:28:28.286 +01:00] [INF] Loaded plugin: "Playback Reporting" "9.0.0.0"
[2021-03-10 02:28:28.287 +01:00] [INF] Loaded plugin: "Reports" "11.0.0.0"
[2021-03-10 02:28:28.288 +01:00] [INF] Loaded plugin: "TMDb Box Sets" "7.0.0.0"
[2021-03-10 02:28:28.290 +01:00] [INF] Loaded plugin: "OMDb" "10.7.0.0"
[2021-03-10 02:28:28.290 +01:00] [INF] Loaded plugin: "MusicBrainz" "10.7.0.0"
[2021-03-10 02:28:28.291 +01:00] [INF] Loaded plugin: "AudioDB" "10.7.0.0"
[2021-03-10 02:28:28.477 +01:00] [INF] Kestrel listening on "Any IP4 Address"
[2021-03-10 02:28:29.434 +01:00] [INF] Running startup tasks
[2021-03-10 02:28:29.435 +01:00] [INF] Retention Task Scheduled!
[2021-03-10 02:28:29.436 +01:00] [INF] TaskCleanDb Loaded
[2021-03-10 02:28:29.440 +01:00] [INF] TaskRunBackup Loaded
[2021-03-10 02:28:29.566 +01:00] [INF] Daily trigger for "Remove Old Sync Data" set to fire at 03/11/2021 00:01, which is 21:32:30.4343004 from now.
[2021-03-10 02:28:29.569 +01:00] [INF] Daily trigger for "Playback Reporting Trim Db" set to fire at 03/11/2021 00:00, which is 21:31:30.4306576 from now.
[2021-03-10 02:28:29.611 +01:00] [INF] Daily trigger for "Extract Chapter Images" set to fire at 03/11/2021 02:00, which is 23:31:30.3883236 from now.
[2021-03-10 02:28:29.769 +01:00] [INF] Found ffmpeg version "4.3.1"
[2021-03-10 02:28:29.818 +01:00] [INF] Available "decoders": ["h264", "h264_qsv", "h264_cuvid", "hevc", "hevc_qsv", "hevc_cuvid", "mpeg2video", "mpeg2_qsv", "mpeg2_cuvid", "mpeg4", "mpeg4_cuvid", "msmpeg4", "vc1_qsv", "vc1_cuvid", "vp8_cuvid", "vp8_qsv", "vp9_cuvid", "vp9_qsv", "aac", "ac3", "flac", "mp3"]
[2021-03-10 02:28:29.843 +01:00] [INF] Available "encoders": ["libx264", "h264_amf", "h264_nvenc", "h264_qsv", "h264_v4l2m2m", "h264_vaapi", "libx265", "hevc_amf", "hevc_nvenc", "hevc_qsv", "hevc_vaapi", "mpeg4", "msmpeg4", "libvpx", "libvpx-vp9", "aac", "ac3", "flac", "libmp3lame", "libopus", "libvorbis", "srt"]
[2021-03-10 02:28:29.864 +01:00] [INF] Available hwaccel types: ["vdpau", "cuda", "vaapi", "qsv", "drm", "opencl"]
[2021-03-10 02:28:29.866 +01:00] [INF] FFmpeg: Custom: "/usr/lib/jellyfin-ffmpeg/ffmpeg"
[2021-03-10 02:28:29.868 +01:00] [INF] ServerId: "bd3e70d255cb4267b503add60dd2138f"
[2021-03-10 02:28:30.125 +01:00] [INF] Registering publisher for "urn:schemas-upnp-org:device:MediaServer:1" on "192.0.2.4/24"
[2021-03-10 02:28:30.162 +01:00] [INF] Executed all pre-startup entry points in 0:00:00.2691583
[2021-03-10 02:28:30.163 +01:00] [INF] Core startup complete
[2021-03-10 02:28:30.165 +01:00] [INF] LibrarySyncNotification Startup...
[2021-03-10 02:28:30.166 +01:00] [INF] UserSyncNotification Startup...
[2021-03-10 02:28:30.167 +01:00] [INF] EventMonitorEntryPoint Running
[2021-03-10 02:28:30.176 +01:00] [INF] Sqlite version: 3.33.0
[2021-03-10 02:28:30.178 +01:00] [INF] Sqlite compiler options: COMPILER=gcc-5.4.0 20160609,DEFAULT_FOREIGN_KEYS,ENABLE_COLUMN_METADATA,ENABLE_FTS3_PARENTHESIS,ENABLE_FTS4,ENABLE_FTS5,ENABLE_JSON1,ENABLE_RTREE,THREADSAFE=1
[2021-03-10 02:28:30.181 +01:00] [INF] Default journal_mode for "/var/lib/jellyfin/data/playback_reporting.db" is "delete"
[2021-03-10 02:28:30.181 +01:00] [INF] Initialize PlaybackActivity Repository
[2021-03-10 02:28:30.182 +01:00] [INF] PlaybackActivity table schema OK
[2021-03-10 02:28:30.182 +01:00] [INF] Expected : "datecreated:datetime|userid:text|itemid:text|itemtype:text|itemname:text|playbackmethod:text|clientname:text|devicename:text|playduration:int"
[2021-03-10 02:28:30.182 +01:00] [INF] Received : "datecreated:datetime|userid:text|itemid:text|itemtype:text|itemname:text|playbackmethod:text|clientname:text|devicename:text|playduration:int"
[2021-03-10 02:28:30.547 +01:00] [INF] Starting NAT discovery
[2021-03-10 02:28:30.550 +01:00] [INF] Watching directory /
[2021-03-10 02:28:30.631 +01:00] [INF] Executed all post-startup entry points in 0:00:00.4679075
[2021-03-10 02:28:30.631 +01:00] [INF] Startup complete 0:00:09.4656256
[2021-03-10 02:28:30.640 +01:00] [INF] Watching directory /
[2021-03-10 02:28:31.171 +01:00] [INF] WS "192.0.2.128" request
[2021-03-10 02:28:32.620 +01:00] [INF] "StartupTrigger" fired for task: "Update Plugins"
[2021-03-10 02:28:32.621 +01:00] [INF] Queuing task "PluginUpdateTask"
[2021-03-10 02:28:32.627 +01:00] [INF] Executing "Update Plugins"
[2021-03-10 02:28:33.746 +01:00] [INF] "Update Plugins" Completed after 0 minute(s) and 1 seconds
[2021-03-10 02:28:33.806 +01:00] [INF] ExecuteQueuedTasks

**Screenshots**
See above

**System (please complete the following information):**
 - Platform: **Linux**
 - Browser: Firefox, PaleMoon
 - Jellyfin Version: 10.7.0

**Additional Context**

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
* Fixes default values provided for invalid config.json files
* Fixes how we load the default config since config.template.json no longer exists (just uses the config available at build time now)

**Issues**
Fixes #2493
Fixes #2494
Fixes #2500

### CODE REVIEW

**Review by ferferga — CHANGES_REQUESTED** [review]
We should log somewhere that the config is invalid imo

File: src/scripts/settings/webSettings.js [hunk 1]
```diff
@@ -55,30 +56,13 @@ async function getConfig() {
...
-
-        if (!response.ok) {
-            throw new Error('network response was not ok');
-        }
-
-        data = await response.json();
-        return data;
-    } catch (error) {
-        console.error('failed to fetch the default web config file:', error);
+        return DefaultConfig;
```
Comment (dmitrylyzo) [inline src/scripts/settings/webSettings.js:59]: It prints 404 error on each `getConfig`.
```suggestion
        data = DefaultConfig;
        return data;
```

File: src/scripts/settings/webSettings.js [hunk 2]
```diff
@@ -126,23 +111,29 @@ const checkDefaultTheme = (themes) => {
...
+        }
+        const themes = Array.isArray(config.themes) ? config.themes : DefaultConfig.themes;
         checkDefaultTheme(themes);
         return themes;
     }).catch(error => {
         console.log('cannot get web config:', error);
         checkDefaultTheme();
-        return [];
+        return DefaultConfig.themes;
     });
```
Comment (dmitrylyzo) [inline src/scripts/settings/webSettings.js:124]: Do we even get into `catch` section?

File: src/scripts/settings/webSettings.js [hunk 3]
```diff
@@ -126,23 +111,29 @@ const checkDefaultTheme = (themes) => {
...
-        return config.plugins;
+        if (!config.plugins) {
+            console.error('web config is invalid, missing plugins:', config);
+        }
+        return config.plugins || DefaultConfig.plugins;
     }).catch(error => {
         console.log('cannot get web config:', error);
-        return [];
+        return DefaultConfig.plugins;
     });
```
Comment (dmitrylyzo) [inline src/scripts/settings/webSettings.js:138]: Do we even get into `catch` section?

File: src/scripts/settings/webSettings.js (hunk 2 above)
Comment (thornbill) [reply src/scripts/settings/webSettings.js:124]: It shouldn't, but better to be on the safe side imo.

**Review by dmitrylyzo — APPROVED** [review]

**Review by ferferga — APPROVED** [review]

**Pull request merged**